### PR TITLE
Use subclass mock maker in atlasdb-commons

### PR DESCRIPTION
**Pull request by schlosna**
## General
**Before this PR**:
There are a bunch of `:atlasdb-commons:test` [flakes on develop](https://app.circleci.com/pipelines/github/palantir/atlasdb?branch=develop) and other branches like https://github.com/palantir/atlasdb/pull/6584 are [blocked on this with odd Mockito failures](https://app.circleci.com/pipelines/github/palantir/atlasdb/12941/workflows/b244564b-b972-442b-ad04-4891715b7d55/jobs/47325/tests)

[CircleCI Flaky tests](https://app.circleci.com/insights/github/palantir/atlasdb/workflows/check/tests?branch=develop)

**After this PR**:
==COMMIT_MSG==
Work around mockito flakes due to BatchingVisitablesTest requiring subclass MockMaker to mock AbstractBatchingVisitable.
==COMMIT_MSG==

**Priority**: high - develop is borked and this is blocking other PRs

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Comment by schlosna** [comment]
These flakes seem to have started around when https://github.com/palantir/atlasdb/pull/6410 merged as https://github.com/palantir/atlasdb/commit/84183ae4ade527e842224fdcf7f03be5f86eca58 . It looks like the PR CI builds passed, but the develop CI build failed. I'm not quite sure how that commit seems to have triggered these Mockito mock maker failures ([it doesn't seem like a classloader conflict as it doesn't look like Mockito uses Caffeine](https://github.com/search?q=repo%3Amockito%2Fmockito+Caffeine&type=code)), but using the same `MockMaker` type within a given module seems to have fixed this. I'd like to find a standalone repro to submit a Mockito bug upstream.

**Review by jeremyk-91 — COMMENTED** [review]
Looks good, thanks for going through this! Just one thing

File: atlasdb-commons/src/test/java/com/palantir/common/concurrent/CheckedRejectionExecutorServiceTest.java [hunk 1]
```diff
@@ -19,31 +19,37 @@
...
 import org.junit.Test;
+import org.mockito.MockMakers;
 
 public class CheckedRejectionExecutorServiceTest {
-    private static final Runnable DO_NOTHING = () -> {};
-    private static final Callable<Integer> RETURN_ONE = () -> 1;
-    private static final RejectedExecutionException REJECTED_EXECUTION_EXCEPTION = new RejectedExecutionException();
+    private final Runnable DO_NOTHING = () -> {};
+    private final Callable<Integer> RETURN_ONE = () -> 1;
+    private final RejectedExecutionException REJECTED_EXECUTION_EXCEPTION = new RejectedExecutionException("test");
```
[the remark this answers is not in the document]
Comment (schlosna) [reply atlasdb-commons/src/test/java/com/palantir/common/concurrent/CheckedRejectionExecutorServiceTest.java:41]: I removed the static when debugging to see if it changed mockito's behavior. Static or instance should work fine. For unit tests I tend to prefer instance fields as their lifetime is confined to individual test case and these are cheap to allocate & create.

Comment (schlosna) [inline atlasdb-commons/src/test/java/com/palantir/common/concurrent/CheckedRejectionExecutorServiceTest.java:41]:
```suggestion
    private static final Runnable DO_NOTHING = () -> {};
    private static final Callable<Integer> RETURN_ONE = () -> 1;
    private static final RejectedExecutionException REJECTED_EXECUTION_EXCEPTION = new RejectedExecutionException("test");
```

**Comment by jeremyk-91** [comment]
need to run spotless :( I'll get to this later today if you're busy

**Comment by schlosna** [comment]
> need to run spotless :( I'll get to this later today if you're busy

need to get nitbot running on OSS at some point

**Comment by jeremyk-91** [comment]
👍

**Pull request merged**